openedx/core/djangoapps/notifications/views.py

```diff
@@ -36,3 +43,90 @@ class CourseEnrollmentListView(generics.ListAPIView):
     def get_queryset(self):
         user = self.request.user
         return CourseEnrollment.objects.filter(user=user, is_active=True)
+
+
+class UserNotificationPreferenceView(APIView):
+    """
+    Supports retrieving and patching the UserNotificationPreference
+    model.
+
+    **Example Requests**
+        GET /api/notifications/configurations/{course_id}
+        PUT /api/notifications/configurations/{course_id}
+
+    **Example Response**:
+    {
+        'id': 1,
+        'course_name': 'testcourse',
+        'course_id': 'course-v1:testorg+testcourse+testrun',
+        'notification_preference_config': {
+            'discussion': {
+                'new_post': {
+                    'web': False,
+                    'push': False,
+                    'email': False,
+                }
+            }
+        }
+    }
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, course_key_string):
+        """
+        Returns notification preference for user for a course.
+
+         Parameters:
+             request (Request): The request object.
+             course_key_string (int): The ID of the course to retrieve notification preference.
+
+         Returns:
+             {
+                'id': 1,
+                'course_name': 'testcourse',
+                'course_id': 'course-v1:testorg+testcourse+testrun',
+                'notification_preference_config': {
+                    'discussion': {
+                        'new_post': {
+                            'web': False,
+                            'push': False,
+                            'email': False,
+                        }
+                    }
+                }
+            }
+         """
+        course_id = CourseKey.from_string(course_key_string)
+        user_notification_preference = NotificationPreference.objects.get(
+            user=request.user,
+            course_id=course_id,
+            is_active=True,
+        )
+        serializer = UserNotificationPreferenceSerializer(user_notification_preference)
+        return Response(serializer.data)
+
+    def put(self, request, course_key_string):
+        """
+        Update an existing user notification preference with the data in the request body.
+
+        Parameters:
+            request (Request): The request object
+            course_key_string (int): The ID of the course of the notification preference to be updated.
+
+        Returns:
+            200: The updated preference, serialized using the UserNotificationPreferenceSerializer
+            404: If the preference does not exist
+            403: If the user does not have permission to update the preference
+            400: Validation error
+        """
+        course_id = CourseKey.from_string(course_key_string)
+        user_notification_preference = NotificationPreference.objects.get(
+            user=request.user,
+            course_id=course_id,
+            is_active=True,
```

Comment:
this field represents whether a Notification Preference is `active` or not. 

For active course, this will be `active` (by default it is `True`). When course is complete or inactive, then this will be `False`.

This is not a field that can be updated by the user. Only field the user can update is the `notification_preference_config`. 
This check is added to the query to fetch a course notification preference for user that is also `active`.